bindings/python/examples/showPointCloud/showPointCloud.py

```diff
@@ -52,134 +53,201 @@ def transform_image(np_image):
-    else :
-        print("Too many arguments provided!")
-        exit(1)
-    
-    status = system.getCameraList(cameras, ip)
-    print("system.getCameraList()", status)
-
-    camera1 = cameras[0]
-
-    status = cameras[0].initialize(config)
-    if not status:
-        print("cameras[0].initialize() failed with status: ", status)
-
-    types = []
-    status = cameras[0].getAvailableFrameTypes(types)
-    if not status:
-        print("system.getAvailableFrameTypes() failed with status: ", status)
-
-    status = cameras[0].setFrameType("lr-qnative")
-    if not status:
-        print("cameras[0].setFrameType() failed with status:", status)
-
-    camDetails = tof.CameraDetails()
-    status = cameras[0].getDetails(camDetails)
-    if not status:
-        print("system.getDetails() failed with status: ", status)
-
-    status = camera1.start()
-    print("camera1.start()", status)
-
-    # Get the first frame for details
-    frame = tof.Frame()
-    status = cameras[0].requestFrame(frame)
-
-    frameDataDetails = tof.FrameDataDetails()
-    status = frame.getDataDetails("depth", frameDataDetails)
-    width = frameDataDetails.width
-    height = frameDataDetails.height
-
-    # Get intrinsic parameters from camera
-    intrinsicParameters = camDetails.intrinsics
-    fx = intrinsicParameters.fx
-    fy = intrinsicParameters.fy
-    cx = intrinsicParameters.cx
-    cy = intrinsicParameters.cy
-    cameraIntrinsics = o3d.camera.PinholeCameraIntrinsic(width, height, fx, fy, cx, cy)
-
-    # Get camera details for frame correction
-    # TO DO: Get the range from camera details when it will be defined
-    camera_range = 4096
-    bitCount = 12
-    max_value_of_AB_pixel = 2 ** bitCount - 1
-    distance_scale_ab = 255.0 / max_value_of_AB_pixel
-    distance_scale = 255.0 / camera_range
-
-    # Create visualizer for depth and ab
-    vis_depth = o3d.visualization.Visualizer()
-    vis_depth.create_window("Depth", 2 * width, 2 * height)
-
-    vis_ab = o3d.visualization.Visualizer()
-    vis_ab.create_window("AB", 2 * width, 2 * height)
-
-    # Create visualizer
-    vis = o3d.visualization.Visualizer()
-    vis.create_window("PointCloud", 1200, 1200)
-    first_time_render_pc = 1
-    point_cloud = o3d.geometry.PointCloud()
-
-    while True:
-        # Capture frame-by-frame
+    ip = ''
+    fileName = args.frame
+    config = args.config
+    if args.ip is not None:
+        ip = 'ip:' + args.ip
+
+    if fileName is None:
+        status = system.getCameraList(cameras, ip)
+        print("system.getCameraList()", status)
+
+        camera1 = cameras[0]
+
+        status = cameras[0].initialize(config)
+        if not status:
+            print("cameras[0].initialize() failed with status: ", status)
+
+        types = []
+        status = cameras[0].getAvailableFrameTypes(types)
+        if not status:
+            print("system.getAvailableFrameTypes() failed with status: ", status)
```

Comment:
Now on dev-5.1.0 branch this method has another name like getAvailableModes

Reply:
Solved